components/feature/tab-collections/src/androidTest/java/mozilla/components/feature/tab/collections/TabCollectionStorageTest.kt

```diff
@@ -0,0 +1,194 @@
+
+        getAllCollections().let { collections ->
+            assertEquals(1, collections.size)
+            assertEquals(0, collections[0].tabs.size)
+
+            storage.addTabsToCollection(collections[0], listOf(
+                Session("https://www.mozilla.org").apply { title = "Mozilla" },
+                Session("https://www.firefox.com").apply { title = "Firefox" }
+            ))
+        }
+
+        getAllCollections().let { collections ->
+            assertEquals(1, collections.size)
+            assertEquals(2, collections[0].tabs.size)
+
+            assertEquals("https://www.mozilla.org", collections[0].tabs[0].url)
+            assertEquals("Mozilla", collections[0].tabs[0].title)
+            assertEquals("https://www.firefox.com", collections[0].tabs[1].url)
+            assertEquals("Firefox", collections[0].tabs[1].title)
+        }
+    }
+
+    @Test
+    fun testRemovingTabsFromCollection() {
+        storage.createCollection("Articles", listOf(
+            Session("https://www.mozilla.org").apply { title = "Mozilla" },
+            Session("https://www.firefox.com").apply { title = "Firefox" }
+        ))
+
+        getAllCollections().let { collections ->
+            assertEquals(1, collections.size)
+            assertEquals(2, collections[0].tabs.size)
+
+            storage.removeTabFromCollection(collections[0], collections[0].tabs[0])
+        }
+
+        getAllCollections().let { collections ->
+            assertEquals(1, collections.size)
+            assertEquals(1, collections[0].tabs.size)
+
+            assertEquals("https://www.firefox.com", collections[0].tabs[0].url)
+            assertEquals("Firefox", collections[0].tabs[0].title)
+        }
+    }
+
+    @Test
+    fun testRenamingCollection() {
+        storage.createCollection("Articles")
+
+        getAllCollections().let { collections ->
+            assertEquals(1, collections.size)
+            storage.renameCollection(collections[0], "Blog Articles")
+        }
+
+        getAllCollections().let { collections ->
+            assertEquals(1, collections.size)
+            assertEquals("Blog Articles", collections[0].title)
+        }
+    }
+
+    @Test
+    fun testRemovingCollection() {
+        storage.createCollection("Articles")
+        storage.createCollection("Recipes")
+
+        getAllCollections().let { collections ->
+            assertEquals(2, collections.size)
+            assertEquals("Recipes", collections[0].title)
+            assertEquals("Articles", collections[1].title)
+
+            storage.removeCollection(collections[0])
+        }
+
+        getAllCollections().let { collections ->
+            assertEquals(1, collections.size)
+
+            assertEquals("Articles", collections[0].title)
+        }
+    }
+
+    private fun getAllCollections(): List<TabCollection> {
+        // Not sure exactly why; but without delaying here `getAllCollections` may not see the latest state.
```

Comment:
Are the user going to experience the same issue ?

Reply:
It seemed fine when testing in the sample browser. I think that may be caused by the test setup where I tried to get a list from the datasource for testing.

Reply:
👍